youtube_dl/extractor/mirror.py

```diff
@@ -0,0 +1,118 @@
+
+        if mobj is None:
+            raise ExtractorError('Video does not exist', expected=True)
+
+        account_id = mobj.group('account_id')
+        video_id = mobj.group('video_id')
+        video_url = mobj.group('video_url')
+
+        player_id = self._search_regex(
+            r'(?:&l?s?quot?;)+playerId(?:&l?s?quot?;)+:(?:&l?s?quot?;)+(?P<player_id>[a-zA-Z0-9]+)(?:&l?s?quot?;)+,',
+            webpage, 'player id', group='player_id'
+        )
+
+        player_url = 'http://players.brightcove.net/%s/%s_default/index.min.js' % (account_id, player_id)
+
+        js_data = self._download_webpage(player_url, '', 'Downloading player info')
+        policy_key = self._search_regex(
+            r'policyKey\s*:\s*[\'"](?P<policy_key>[^\'"]+)',
+            js_data, 'policy key', group='policy_key'
+        )
+
+        api_url = 'https://edge.api.brightcove.com/playback/v1/accounts/%s/videos/%s'
+        data = self._download_json(
+            api_url % (account_id, video_id),
+            video_id, headers={
+                'Referer': url,
+                'Accept': 'application/json;pk=%s' % policy_key,
+            }
+        )
+
+        formats = []
+        for source in data.get('sources'):
+            if source.get('app_name') is not None:
+                continue
+            if source.get('type') == 'application/x-mpegURL':
+                formats.append({
+                    'url': source.get('src'),
+                    'manifest_url': source.get('src'),
+                    'container': source.get('container').lower(),
+                    'format': 'mp4',
+                    'ext': 'mp4',
+                    'vcodec': source.get('codec'.lower()),
+                    'protocol': 'm3u8_native'
+                })
+                continue
+
+            formats.append({
+                'url': source.get('src'),
+                'tbr': source.get('avg_bitrate'),
+                'width': source.get('width'),
+                'height': source.get('height'),
+                'resolution': '%sx%s' % (source.get('width'), source.get('height')),
+                'filesize': source.get('size'),
+                'vcodec': source.get('codec').lower(),
+                'ext': source.get('container').lower(),
+            })
+
+        formats = sorted(formats, key=lambda k: k.get('height', True))
+        return {
+            'id': video_id,
+            'title': data.get('name'),
+            'formats': formats,
+            'url': video_url,
+            'ext': determine_ext(video_url, 'mp4'),
+            'duration': data.get('duration'),
+            'tags': data.get('tags'),
+            'timestamp': parse_iso8601(data.get('published_at')),
+            'uploader': data.get('custom_fields').get('uploader'),
+            'thumbnails': [{'url': x.get('src')} for x in data.get('thumbnail_sources')]
+        }
```

Comment:
I delegated extraction to brightcove and provided a fallback in case brightcave extractor fails